### PR TITLE
Dx 1597 voice callback objects in spec

### PR DESCRIPTION
**Pull request by jmulford-bw**
## For the Committer

All PRs on this repo that change any API sources of truth (markdown files, OpenAPI specs) require a changelog added to the `site/src/pages/changelog.md` file. If this PR does not require changelog updates, you need to add the `no-changelog` tag to this PR before opening it.

Please confirm that you have either updated `site/src/pages/changelog.md` or added the `no-changelog` tag.

### CODE REVIEW

**Comment by mhossomi** [comment]
These two `*Callback` objects are very generic versions of all event types that can happen (e.g. answer, initiate, gather, disconnect etc), is that the intention? They aren't too different but, some fields appear only in some and not in others.

**Comment by jmulford-bw** [comment]
> These two `*Callback` objects are very generic versions of all event types that can happen (e.g. answer, initiate, gather, disconnect etc), is that the intention? They aren't too different but, some fields appear only in some and not in others.

Yes. Would it make more sense to divide them up further, or consolidate them into one?

**Comment by mhossomi** [comment]
> > These two `*Callback` objects are very generic versions of all event types that can happen (e.g. answer, initiate, gather, disconnect etc), is that the intention? They aren't too different but, some fields appear only in some and not in others.
> 
> Yes. Would it make more sense to divide them up further, or consolidate them into one?

Consolidating into one is bad in my opinion. As is, users will have to refer to the documentation to know which fields they can expect for each event type. Personally, I'm ok with that, but might want to mention it somewhere along with this spec for unexperienced users?

File: site/specs-source/voice.json [hunk 1]
```diff
@@ -3250,6 +3250,191 @@
             "type": "string"
           }
         }
+      },
+      "CallCallback": {
```
Comment (mhossomi) [inline site/specs-source/voice.json:3254]: Are my eyes tricking me or `recordingId` is missing here?

Comment (jmulford-bw) [reply site/specs-source/voice.json:3254]: It's on line 3314

**Comment by jmulford-bw** [comment]
> > > These two `*Callback` objects are very generic versions of all event types that can happen (e.g. answer, initiate, gather, disconnect etc), is that the intention? They aren't too different but, some fields appear only in some and not in others.
> > 
> > 
> > Yes. Would it make more sense to divide them up further, or consolidate them into one?
> 
> Consolidating into one is bad in my opinion. As is, users will have to refer to the documentation to know which fields they can expect for each event type. Personally, I'm ok with that, but might want to mention it somewhere along with this spec for unexperienced users?

Awesome sounds good. I think we can add a `description` attribute to the objects themselves to clarify this a little bit

File: site/specs-source/voice.json (hunk 1 above)
Comment (mhossomi) [reply site/specs-source/voice.json:3254]: 🙈

**Review by mhossomi — APPROVED** [review]

**Pull request merged**